node/CMakeLists.txt

```diff
@@ -101,7 +101,7 @@ set(BUILD_FLAGS "${BUILD_FLAGS} -Wold-style-cast")
 set(BUILD_FLAGS "${BUILD_FLAGS} -Woverloaded-virtual")
 set(BUILD_FLAGS "${BUILD_FLAGS} -Wswitch")
 set(BUILD_FLAGS "${BUILD_FLAGS} -Wunreachable-code")
-set(BUILD_FLAGS "${BUILD_FLAGS} -pedantic -pedantic-errors")
+set(BUILD_FLAGS "${BUILD_FLAGS} -pedantic")
```

Comment:
Why did you remove `-pedantic-errors`?

Reply:
To make lambda capture assignment possible, it is a warning with pedantic option for current gcc version. And assignment is needed to pass flag_sentry_t to the asyc. completion handlers without polluting the lambda code by extra variable.

Reply:
Looks like you don't use this feature in current version of PR

Reply:
Need to switch '-pedantic-errors' to allow to compile moved lambda captures:
```cpp
socket_->async_connect(std::move(endpoint), [self, callback=std::move(callback)](const std::error_code& ec) {
// ...
});
```

Reply:
Oh, sorry, I missed that.